src/GraphQL/Utilities/Visitors/FieldTypeDefaultArgumentsVisitor.cs

```diff
@@ -0,0 +1,29 @@
+using GraphQL.Execution;
+using GraphQL.Types;
+
+namespace GraphQL.Utilities
+{
+    /// <summary>
+    /// Sets <see cref="FieldType.DefaultArguments"/> for each <see cref="FieldType"/>.
+    /// </summary>
+    public sealed class FieldTypeDefaultArgumentsVisitor : BaseSchemaNodeVisitor
+    {
+        /// <summary>
+        /// Returns a static instance of the <see cref="FieldTypeDefaultArgumentsVisitor"/> class.
+        /// </summary>
+        public static readonly FieldTypeDefaultArgumentsVisitor Instance = new();
+
+        private FieldTypeDefaultArgumentsVisitor()
+        {
+        }
+
+        /// <inheritdoc/>
+        public override void VisitObjectFieldDefinition(FieldType field, IObjectGraphType type, ISchema schema)
+        {
+            if (field.Arguments?.Count > 0)
+            {
+                field.DefaultArguments = field.Arguments.ToDictionary(arg => arg.Name, arg => new ArgumentValue(arg.DefaultValue, ArgumentSource.FieldDefault));
```

Comment:
Two questions.
1. override `VisitInterfaceFieldDefinition` as well?
2. Include only args where `arg.DefaultValue` is not null? As I understand now `field.DefaultArguments` will always contain all arguments regardless whether it has any default value or not.

Reply:
Or if you do that intentionally then I would clarify what _Default_ Arguments means in that context.

Reply:
> override VisitInterfaceFieldDefinition as well?

Good catch, but unnecessary in the end.  I wrote a test which proved that due to #3417 it will not be used.  I added it anyway though.

Reply:
> Include only args where `arg.DefaultValue` is not null? As I understand now `field.DefaultArguments` will always contain all arguments regardless whether it has any default value or not.

No; this would complicate the implementation with perhaps zero benefit.  Could be slower.  Default field arguments still need to be added to the dictionary, and cloning from the default argument values doesn't provide a material benefit, since the default value is already in native form.

Reply:
> Or if you do that intentionally then I would clarify what _Default_ Arguments means in that context.

Can you help here with comments/naming/etc as you see fit?   As I think you understand, it is a dictionary of argument values with all values set to the field default value (`null` or otherwise), used to pass to field resolvers when no arguments are supplied to the field within the query document.  Note: non-null constraints are ignored for this purpose, although if there are any non-null field arguments, then (assuming validation has not been disabled) it should be impossible to need this `DefaultArguments` value.

Reply:
>  it is necessary to provide a complete list of arguments to field resolvers, regardless of whether the argument contains a default value or is null.

To align behavior with `ExecutionHelper.GetArguments` that returns dictionary of all argument values?

Reply:
Yes (I was specifically thinking of `IResolveFieldContext.Arguments`, which just calls `ExecutionHelper.GetArguments`.)

Reply:
> Can you help here with comments/naming/etc as you see fit? As I think you understand, it is a dictionary of argument values with all values set to the field default value (null or otherwise), used to pass to field resolvers when no arguments are supplied to the field within the query document. 

Uhhh, naming is hard. I look at the code trying to feel the essence of this property and it seems to me more and more that it is more likely a sort of optimizarion. It does not carry an own semantic - just the same arguments as `field.Arguments` but grouped by name (selection of default values from arguments of course). 

By the way, it seems to be worth making it internal.

Reply:
And **IF** it can be internal then I'll suggest to name it as `AllArgumentsDefaultValues`

Reply:
`DefaultArguments` sounds ands feels like another concept. When I was reading the code for the first time, it confused me - all arguments vs only those that have defaults.

Reply:
How about `DefaultArgumentValues`?

Reply:
Maybe have it `public` with `internal set;` ?

Reply:
> DefaultArgumentValues

OK. I would not take this property into public API surface. IMO it's true implementation detail that we want to use to optimize allocations. We can do without this property at all. It serves like data cache.

Reply:
Perhaps `ArgumentsDefaultValues`?

What if we move the property underneath `QueryArguments` so it is less 'visible'?  Perhaps `QueryArguments.DefaultValues` or `DefaultValuesCollection`.

Reply:
Ok, I'm fine with it remaining `internal` until requested otherwise.

Well, after the rework, it would sorta be necessary unless using `ReadonlyResolveFieldContext` (directly or indirectly).  We surely will need to allow custom validation rules to pull the argument values dictionary, so if nothing else, we will need to add an extension method or something.

Reply:
> What if we move the property underneath QueryArguments so it is less 'visible'? Perhaps QueryArguments.DefaultValues or DefaultValuesCollection.

This changes nothing IMO. It would be more simple to keep it in `FieldType` as internal property.

Reply:
Ok done.